core/src/main/java/nl/nn/adapterframework/filesystem/FileSystemActor.java

```diff
@@ -186,28 +175,47 @@ public void configure(FS fileSystem, ParameterList parameterList, INamedObject o
 			}
 		}
 	}
+
+	private void checkConfiguration(String action) throws ConfigurationException {
+		if (!actions.contains(action))
+			throw new ConfigurationException(ClassUtils.nameOf(owner)+" ["+owner.getName()+"]: unknown or invalid action [" + action + "] supported actions are " + actions.toString() + "");
+
+		if (action.equals(ACTION_READ2)) {
+			ConfigurationWarnings.add(owner, log, "action ["+ACTION_READ2+"] has been replaced with ["+ACTION_READ1+"]");
+			setAction(ACTION_READ1);
+		}
+		if (action.equals(ACTION_WRITE2)) {
+			ConfigurationWarnings.add(owner, log, "action ["+ACTION_WRITE2+"] has been replaced with ["+ACTION_WRITE1+"]");
+			setAction(ACTION_WRITE1);
+		}
```

Comment:
This part must be at configure(), to support legacy action names. We do not want/need to support those names for dynamic actions, and therefore the code can be moved back to configure(). It cannot stay here, because it sets the action via setAction().